S3/FileLists.py

```diff
@@ -60,24 +60,29 @@ def handle_exclude_include_walk_dir(root, dirname):
     Should this root/dirname directory be excluded? (otherwise included by default)
     Exclude dir matches in the current directory
     This prevents us from recursing down trees we know we want to ignore
-    return True for including, and False for excluding
+    return True for excluding, and False for including
     '''
     cfg = Config()
+    # python versions end their patterns (from globs) differently, test for different styles; check python3.6+ styles first
+    directory_patterns = [u'/)$', u'/)\\Z', u'\\/$', u'\\/\\Z(?ms)']
```

Comment:
A last small thing, could you transform that into a "tuple" instead of a list as it will not be dynamically changed?